Library/OcAppleKernelLib/CommonPatches.c

```diff
@@ -1064,6 +1064,126 @@ PatchIncreasePciBarSize (
   return Status;
 }
 
+STATIC
+CONST UINT8
+mSerialDevicePmioFind[] = {
+  0x66, 0xBA, 0xF8, 0x03  ///< mov dx, 0x03F[8-9A-F]
+};
+STATIC_ASSERT (sizeof (mSerialDevicePmioFind) == 4, "Unsupported mSerialDevicePmioFind");
+
+STATIC
+CONST UINT8
+mSerialDevicePmioMask[] = {
+  0xFF, 0xFF, 0xF8, 0xFF
+};
+STATIC_ASSERT (sizeof (mSerialDevicePmioMask) == 4, "Unsupported mSerialDevicePmioMask");
+
+STATIC
+UINT8
+mSerialDevicePmioReplace[] = {
+  0x66, 0xBA, 0x00, 0x00  ///< mov dx, whatever ; To be set by PatchSetPciSerialDeviceRegisterBase()
+};
+STATIC_ASSERT (sizeof (mSerialDevicePmioReplace) == 4, "Unsupported mSerialDevicePmioReplace");
+
+STATIC
+PATCHER_GENERIC_PATCH
+mCustomPciSerialDevicePmioPatch = {
+  .Comment     = DEBUG_POINTER ("CustomPciSerialDevicePmio"),
+  .Base        = "_serial_init",
+  .Find        = mSerialDevicePmioFind,
+  .Mask        = mSerialDevicePmioMask,
+  .Replace     = mSerialDevicePmioReplace,
+  .ReplaceMask = mSerialDevicePmioMask,
+  .Size        = sizeof (mSerialDevicePmioFind),
+  .Count       = 0,
+  .Skip        = 0,
+  .Limit       = 1024
+};
+
+STATIC
+CONST UINT8
+mSerialDeviceMmioFind[] = {
+  0x00, 0x40, 0x03, 0xFE  ///< mov whatever, 0xFE0340xx
+};
+STATIC_ASSERT (sizeof (mSerialDeviceMmioFind) == 4, "Unsupported mSerialDeviceMmioFind");
```

Comment:
This looks odd, why must it be exactly 4? Maybe just ASSERT Find, Mask, and Replace are of equal length, whatever it is?

Reply:
https://github.com/acidanthera/OpenCorePkg/pull/331/commits/d42e4665b129677bdf8766b76e8153e37da8d0b3